AsyncDisplayKit/ASTextNode.mm

```diff
@@ -749,32 +785,28 @@ - (NSArray *)_rectsForTextRange:(NSRange)textRange measureOption:(ASTextKitRende
   return adjustedRects;
 }
 
-- (NSArray *)rectsForTextRange:(NSRange)textRange
-{
-  return [self _rectsForTextRange:textRange measureOption:ASTextKitRendererMeasureOptionCapHeight];
-}
-
-- (NSArray *)highlightRectsForTextRange:(NSRange)textRange
-{
-  return [self _rectsForTextRange:textRange measureOption:ASTextKitRendererMeasureOptionBlock];
-}
-
 - (CGRect)trailingRect
 {
+  std::lock_guard<std::recursive_mutex> l(_textLock);
+  
   CGRect rect = [[self _renderer] trailingRect];
-  return [self.class _adjustRendererRect:rect forShadowPadding:self.shadowPadding];
+  return ASTextNodeAdjustRenderRectForShadowPadding(rect, self.shadowPadding);
 }
 
 - (CGRect)frameForTextRange:(NSRange)textRange
 {
+  std::lock_guard<std::recursive_mutex> l(_textLock);
```

Comment:
I don't think this needs a lock. self _renderer is locked so you're guaranteed an atomic access to the renderer, then the subsequent operations should be OK without locking since ASTextNodeRenderer should be threadsafe.